File: packages/agent_security/android/src/main/kotlin/com/chapter2/agentkit/models/WorldIdCredentialType.kt
package com.chapter2.agentkit.models

public sealed class WorldIdCredentialType(
    val codeString: String,
    val isBiometricVerified: Boolean
) {
    data object Credential11Selfie : WorldIdCredentialType("11", isBiometricVerified = true)
    data object Orb : WorldIdCredentialType("orb", isBiometricVerified = true)
    data object Device : WorldIdCredentialType("device", isBiometricVerified = false)

    companion object {
        fun fromString(raw: String): WorldIdCredentialType? = when (raw.lowercase().trim()) {
            "11", "selfie" -> Credential11Selfie
            "orb" -> Orb
            "device", "0" -> Device
            else -> null
        }
    }
}

File: packages/agent_security/ios/Classes/Models/WorldIdCredentialType.swift
import Foundation

public enum WorldIdCredentialType: Equatable, Hashable, Codable, Sendable {
    case credential11Selfie
    case orb
    case device

    public var codeString: String {
        switch self {
        case .credential11Selfie:
            return "11"
        case .orb:
            return "orb"
        case .device:
            return "device"
        }
    }

    public var isBiometricVerified: Bool {
        switch self {
        case .credential11Selfie, .orb:
            return true
        case .device:
            return false
        }
    }

    public static func from(raw: String) -> WorldIdCredentialType? {
        switch raw.lowercased() {
        case "11", "selfie":
            return .credential11Selfie
        case "orb":
            return .orb
        case "device", "0":
            return .device
        default:
            return nil
        }
    }
}
